packages/frontend/src/features/metricsCatalog/components/visualization/MetricPeekDatePicker.tsx

```diff
@@ -77,11 +83,26 @@ export const MetricPeekDatePicker: FC<Props> = ({
         });
     }, [organizationUuid, projectUuid, track]);
 
-    const matchingPresetLabel = getMatchingPresetLabel(dateRange, timeInterval);
+    const lastStableMatchingPresetLabel = useRef(
```

Comment:
could use useState, but useRef is actually a simpler solution that doesn't trigger re-renders